### PR TITLE
Generalize library structure for Tridiagonalization

### PR DESCRIPTION
**Pull request by albestro**
Preparatory work for next Tridiagonalization PR to make it accept input matrix on `Device::GPU`.
Computation will still be on `Backend::MC`.

### CODE REVIEW

File: include/dlaf/eigensolver/band_to_tridiag/api.h [hunk 1]
```diff
@@ -7,18 +7,18 @@
...
 #include "dlaf/matrix/matrix.h"
 #include "dlaf/types.h"
 
-namespace dlaf {
-namespace eigensolver {
+namespace dlaf::eigensolver {
 
-template <class T, Device device>
+template <class T, Device D>
 struct ReturnTridiagType {
```
Comment (msimberg) [inline include/dlaf/eigensolver/band_to_tridiag/api.h:19]: Not your changes, but should this also be renamed `TridiagResult` like for the eigensolver result type?

**Review by msimberg — APPROVED** [review]

File: include/dlaf/eigensolver/band_to_tridiag/api.h (hunk 1 above)
Comment (albestro) [reply include/dlaf/eigensolver/band_to_tridiag/api.h:19]: Yes, I can do it here definitely! @rasolca do you agree?

**Review by rasolca — APPROVED** [review]

**Comment by rasolca** [comment]
bors try

**Comment by rasolca** [comment]
bors ping

**Comment by rasolca** [comment]
bors try

**Pull request merged**